app/controllers/reports_controller.rb

```diff
@@ -0,0 +1,44 @@
+class ReportsController < ApplicationController
+
+
+  def index;
+    @categories = Category.order(:id).all
+  end
+
+  def create;
```

Comment:
A i nie wydaje mi się, że `create` jest tutaj odpowiednią akcją. Nie tworzymy nic w bazie, jedyne co robimy to wyszukujemy specyficznych customerów, więc jak dla mnie to raczej customowa akcja powinna być.

Reply:
W widoku index formularz ma button, który kieruje do metody create. Nie wiem jak to zmienić.

Reply:
Podsyłam diffa:
```
diff --git a/app/controllers/reports_controller.rb b/app/controllers/reports_controller.rb
index aa40b45..c3ce1be 100644
--- a/app/controllers/reports_controller.rb
+++ b/app/controllers/reports_controller.rb
@@ -4,7 +4,7 @@ class ReportsController < ApplicationController
     @categories = Category.order(:id).all
   end
 
-  def create;
+  def customers_without_orders
     set_parameters[:report]
     report = CustomersWithoutOrders.new(
         start_date: @parameters[:start_date],
diff --git a/app/views/reports/create.html.haml b/app/views/reports/create.html.haml
deleted file mode 100644
index 9fc9364..0000000
--- a/app/views/reports/create.html.haml
+++ /dev/null
@@ -1,13 +0,0 @@
-%div{ class: "container" }
-  %div{ class: "text-center bottom-margin" }
-    %h1
-      Reports
-    = link_to "Back to all reports", reports_path, class: "btn btn-info"
-  %div{ class: "text-center bottom-margin" }
-    %h4
-      ="Lista klientów, którzy nie złożyli żadnego zamówienia na produkt z kategorii '#{@parameters[:category]}' w okresie od #{@parameters[:start_date]} do #{@parameters[:end_date]}."
-    - @customers.each do |customer|
-      %div{ class: "row" }
-        %div{ class: "col-xl" }
-          %table{ class: "table table-striped" }
-            = render "customers_list", customer: customer
diff --git a/app/views/reports/index.html.haml b/app/views/reports/index.html.haml
index b3a0c20..d7640f2 100644
--- a/app/views/reports/index.html.haml
+++ b/app/views/reports/index.html.haml
@@ -10,7 +10,7 @@
         1. Lista klientów, którzy nie złożyli żadnego zamówienia
       %h4
         Parametry raportu:
-        = simple_form_for :report do |form|
+        = simple_form_for :report, url: customers_without_orders_path, method: :get do |form|
           %div{ class: "row" }
             %div{ class: "col-xl" }
               %table{ class: "table table-striped" }
diff --git a/config/routes.rb b/config/routes.rb
index 2ed0ee0..d396076 100644
--- a/config/routes.rb
+++ b/config/routes.rb
@@ -4,5 +4,6 @@ Rails.application.routes.draw do
   resources :orders
   resources :products
   resources :reports
+  get "customers_without_orders", to: "reports#customers_without_orders"
   match "*path", to: "main#index", via: :all
 end
```